### DISCUSSION TITLE
Desktop main window `IsVisible` is ignored

### DISCUSSION THREAD

**Discussion by lostmsu** [AvaloniaUI/Avalonia#12863] [Q&A]
**To Reproduce**

1. In a simple app, set main window's `IsVisible` to `false` in XAML, constructor, or initializer (e.g. `new Window{ IsVisible = false }`)

**Expected behavior**

Window is created but is invisible until `Show()` or `IsVisible = true`

**Actual behavior**

Window shows up

**Desktop (please complete the following information):**

- OS: Windows 11 22H2
- Version 11.0.4

**Comment by kekekeks** [comment]
`desktopLifetime.MainWindow` gets shown automatically when control flow leaves `OnFrameworkInitializationCompleted`. Don't set that property from `OnFrameworkInitializationCompleted` if you need to manually control it.

**Reply by lostmsu** [reply]
@kekekeks is there a reason to do that? Why should the `IsVisible` property not be respected?

Are there any scenarios in which having `IsVisible` purposefully `false` and still wanting main window show up when assigned specifically in `OnFrameworkInitializationCompleted` makes sense?

**Comment by timunie** [comment]
if you do what @kekekeks suggested, remember to set shutdown mode to explicit. 

```cs
if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
{
    // Set this, otherwise App will close immediately 
    desktop.ShutdownMode = ShutdownMode.OnExplicitShutdown;
    
    // Don't set this, if you don't want to show a MainWindow on startup
    // desktop.MainWindow = new MainWindow
    // {
    //     DataContext = new MainViewModel()
    // };
}
```

**Reply by lostmsu** [reply]
I don't think this is actually true in practice. Before trying the above I was delaying creating main window until user clicks tray icon, and I had `ShutdownMode.OnMainWindowClose`, and the app did not close immediately.

**Comment by timunie** [comment]
@lostmsu I'll convert this into a Q&A for now. If you still see an issue, please file a minimal sample.
